### PR TITLE
chore: update api with new format & kubebuilder

### PR DESCRIPTION
**Pull request by MatousJobanek**
result of codeready-toolchain/api#237
 * small changes in go.mod so we can use the new controller-tools 0.4.1 together with client-go 0.18.3
 * update all imports to api repo/types

### CODE REVIEW

File: pkg/client/client_test.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ import (
 	"testing"
 	"time"
 
-	"github.com/codeready-toolchain/api/pkg/apis"
+	"github.com/codeready-toolchain/api/api/v1alpha1"
```
Comment (xcoulon) [inline pkg/client/client_test.go:11]: Could we all agree to use `toolchainv1alpha1` as an alias for `github.com/codeready-toolchain/api/api/v1alpha1` in all our codebase? 
IMHO, `v1alpha1` alone is a bit meaningless. Also, we've been doing something similar for other packages we import. Eg:
```
metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```

**Review by xcoulon — APPROVED** [review]
I left a comment, but nothing blocking. We can discuss about it and/or address it later

File: pkg/client/client_test.go (hunk 1 above)
Comment (MatousJobanek) [reply pkg/client/client_test.go]: sure - done in f9d3333

**Review by alexeykazakov — APPROVED** [review]

File: pkg/client/client_test.go (hunk 1 above)
Comment (xcoulon) [reply pkg/client/client_test.go]: awesome, thanks!

**Pull request merged**